File: SOLUTIONS/LEETCODE/EASY/isPalindrome/isPalindrome.py
def isPalindrome(s) :
    check = ''.join([c.lower(), for c in s if c.isalnum()])
    pali = check [::-1]
    return check == pali


    # ANOTHER SOLUTION THAT ELUDES THE REGEX EXPRESSION

def isPalindrome(s):
    # clean the string to lowercase and return only alphanumeric
    cleaned = [] 
    for char in s:
        if char.isalnum:
            cleaned.append(char.lower())
    check = ''.join(cleaned)

    pali = check[::1]
    return pali == check

File: SOLUTIONS/LEETCODE/isPalindrome/isPalindrome.js
function isPalindrome(s){
    // to lowercase first before we use regex to expunge all signs and exclamations, then join
    let check = s.toLowercase().replace(/[^a-z0-9]/gi, '').join('')
    // pali reverses the value of check
    let pali = check.split('').reverse().join('')

    if(pali === check){
        return true
    }
    return false
}

/* 
Use Case
Input: "A man, a plan, a canal: Panama"
Cleaned: "amanaplanacanalpanama"
Reversed: "amanaplanacanalpanama"
Returns: True 
*/